internal/bpool/bpool.go

```diff
@@ -34,6 +34,22 @@ func NewBytePoolCap(maxSize int, width int, capwidth int) (bp *BytePoolCap) {
 	}
 }
 
+// Populate - populates and pre-warms the byte pool, this function is non-blocking.
+func (bp *BytePoolCap) Populate() {
+	go func() {
+		for i := 0; i < cap(bp.c); i++ {
+			var b []byte
+			// create new buffer
+			if bp.wcap > 0 {
+				b = make([]byte, bp.w, bp.wcap)
+			} else {
+				b = make([]byte, bp.w)
+			}
+			bp.Put(b)
+		}
+	}()
+}
```

Comment:
No need for goroutines - you are wastly overestimatting the cost of allocations.

You can allocate a single slice and cut it into N slices.

Reply:
> You can allocate a single slice and cut it into N slices. Even easier, use [AllocAligned](https://pkg.go.dev/github.com/klauspost/reedsolomon#AllocAligned), which also gives all slices and also makes them that are cacheline size aligned.

That requires a complete change in the buffers @klauspost i.e [][]byte vs []byte